pkg/morph/client/notary.go

```diff
@@ -411,26 +415,95 @@ func (c *Client) NotarySignAndInvokeTX(mainTx *transaction.Transaction) error {
+	// thus, fill it from the main transaction's signers list.
+	//
+	// TODO: @roman-khimov, this list of signers is used only for notary Actor constructor,
+	//  after that it's useless. Thus, I'd skip the signers parsing down below and create
+	//  some fake account instead to pass the Actor constructor constraint.
+	s := make([]actor.SignerAccount, 0, 3)
+	s = append(s, []actor.SignerAccount{
+		{
+			// Proxy contract that will pay for the execution.
+			Signer:  mainTx.Signers[0],
+			Account: notary.FakeContractAccount(mainTx.Signers[0].Account),
+		},
+		{
+			// Inner ring multisignature.
+			Signer:  mainTx.Signers[1],
+			Account: multiaddrAccount,
+		},
+	}...)
+
+	if len(mainTx.Signers) > 3 {
+		// Invoker signature (simple signature account of storage node is expected).
+		var acc *wallet.Account
+		script := mainTx.Scripts[2].VerificationScript
+		if len(script) == 0 {
+			acc = notary.FakeContractAccount(mainTx.Signers[2].Account)
+		} else {
+			pubBytes, ok := vm.ParseSignatureContract(script)
+			if ok {
+				pub, err := keys.NewPublicKeyFromBytes(pubBytes, elliptic.P256())
+				if err != nil {
+					return fmt.Errorf("failed to parse verification script of signer #2: invalid public key: %w", err)
+				}
+				acc = notary.FakeSimpleAccount(pub)
+			} else {
+				m, pubsBytes, ok := vm.ParseMultiSigContract(script)
+				if !ok {
+					return errors.New("failed to parse verification script of signer #2: unknown witness type")
+				}
+				pubs := make(keys.PublicKeys, len(pubsBytes))
+				for i := range pubs {
+					pubs[i], err = keys.NewPublicKeyFromBytes(pubsBytes[i], elliptic.P256())
+					if err != nil {
+						return fmt.Errorf("failed to parse verification script of signer #2: invalid public key #%d: %w", i, err)
+					}
+				}
+				acc, err = notary.FakeMultisigAccount(m, pubs)
+				if err != nil {
+					return fmt.Errorf("failed to create fake account for signer #2: %w", err)
+				}
+			}
+		}
+		s = append(s, actor.SignerAccount{
+			Signer:  mainTx.Signers[2],
+			Account: acc,
+		})
+	}
+
+	nAct, err := notary.NewActor(c.client, s, c.acc)
+	if err != nil {
+		return err
+	}
+
+	// Sign exactly the same transaction we've got from the received Notary request.
+	err = nAct.Sign(mainTx)
+	if err != nil {
+		return fmt.Errorf("faield to sign notary request: %w", err)
+	}
+
+	// TODO: use c.notary.fallbackTime to set the fallback's NotValidBefore attribute.
+	//
+	// @roman-khimov, Currently Actor doesn't allow to customize fallback's NVB, see the
+	// https://github.com/nspcc-dev/neo-go/blob/7bc9b6e3236e4e89aea3ef86240746f3d17e591f/pkg/rpcclient/notary/actor.go#L278
+	// We may either adjust the Actor or create fallback manually right here.
+	// Honestly, I'd suggest to completely remove `fallbackTime` from the neofs-node's
+	// code/config in this PR, it's just not needed and not used anywhere except notary request creation.
+	// ACK removal or deprecation of `fallbackTime`?
```

Comment:
Yeah, it makes zero sense as a configuration option.

Reply:
Removed.